src/main/webapp/features/screens/game/dashboard/gameMap/ct-game-map.js

```diff
@@ -31,7 +43,29 @@ angular.module('catan')
                         var edgeId = angular.element(event.target).closest(DrawMapService.EDGE_SELECTOR).attr('edge-id');
                         SelectMapObjectService.select('edge', edgeId);
                     });
+
                 }
             };
 
+            //TODO: move that code to some helper?
+            function highlightHexes(element, map, dice, timeout) {
+                var hexesToHighlight = map.hexes.filter(function(hex) {
+                    return (dice === 7) ? hex.robbed : !hex.robbed && hex.dice === dice;
+                });
+
+                $timeout.cancel(timeout);
+                element.find(DrawMapService.HEX_SELECTOR).removeClass(HEXES_HIGHLIGHT_CLASS);
+
+                var elementsToHighlight = angular.element([]);
+                hexesToHighlight.forEach(function(hex) {
+                    elementsToHighlight = elementsToHighlight.add(DrawMapService.HEX_SELECTOR + "[hex-id="+hex.hexId+"]");
```

Comment:
why do you need set value of elementsToHighlight again and again? when you call add() method. isn't it extends content of elementsToHighlight ?